File: assets/bin/Data/Managed/Assembly-CSharp/GetSocialSdk.Core/NotificationsQuery.cs
using System;
using UnityEngine;

namespace GetSocialSdk.Core
{
	public sealed class NotificationsQuery : IConvertableToNative
	{
		public enum Filter
		{
			NoFilter,
			Older,
			Newer
		}

		private static readonly Notification.NotificationTypes[] AllTypes = new Notification.NotificationTypes[0];

		private readonly bool? _isRead;

		private Notification.NotificationTypes[] _types = AllTypes;

		private Filter _filter;

		private string _notificationId;

		private int _limit;

		private NotificationsQuery(bool? isRead)
		{
			_isRead = isRead;
		}

		public static NotificationsQuery ReadAndUnread()
		{
			return new NotificationsQuery(null);
		}

		public static NotificationsQuery Read()
		{
			return new NotificationsQuery(true);
		}

		public static NotificationsQuery Unread()
		{
			return new NotificationsQuery(false);
		}

		public NotificationsQuery OfAllTypes()
		{
			_types = AllTypes;
			return this;
		}

		public NotificationsQuery OfTypes(params Notification.NotificationTypes[] types)
		{
			_types = types;
			return this;
		}

		public NotificationsQuery WithLimit(int limit)
		{
			_limit = limit;
			return this;
		}

		public NotificationsQuery WithFilter(Filter filter, string notificationId)
		{
			_filter = filter;
			_notificationId = notificationId;
			return this;
		}

		public AndroidJavaObject ToAjo()
		{
			//IL_0005: Unknown result type (might be due to invalid IL or missing references)
			//IL_000a: Expected O, but got Unknown
			AndroidJavaClass ajo = new AndroidJavaClass("im.getsocial.sdk.pushnotifications.NotificationsQuery");
			AndroidJavaObject val = (!_isRead.HasValue) ? ajo.CallStaticAJO("readAndUnread") : ajo.CallStaticAJO((!_isRead.Value) ? "unread" : "read");
			val.CallAJO("withLimit", _limit);
			if (_types.Length > 0)
			{
				int[] array = Array.ConvertAll(_types, (Notification.NotificationTypes type) => (int)type);
				val.CallAJO("ofTypes", array);
			}
			if (_filter != 0)
			{
				val.CallAJO("withFilter", _filter.ToAndroidJavaObject(), _notificationId);
			}
			return val;
		}
	}
}

File: sources/im/getsocial/sdk/pushnotifications/NotificationsQuery.java
package im.getsocial.sdk.pushnotifications;

import im.getsocial.sdk.pushnotifications.p067a.p103b.upgqDBbsrL;

public final class NotificationsQuery {
    public static final int DEFAULT_LIMIT = 20;
    /* renamed from: b */
    private static final int[] f2458b = new int[0];
    /* renamed from: a */
    final upgqDBbsrL f2459a;

    public enum Filter {
        NO_FILTER,
        OLDER,
        NEWER
    }

    private NotificationsQuery(Boolean bool) {
        this.f2459a = new upgqDBbsrL(bool);
    }

    public static NotificationsQuery read() {
        return new NotificationsQuery(Boolean.valueOf(true));
    }

    public static NotificationsQuery readAndUnread() {
        return new NotificationsQuery(null);
    }

    public static NotificationsQuery unread() {
        return new NotificationsQuery(Boolean.valueOf(false));
    }

    public final NotificationsQuery ofAllTypes() {
        this.f2459a.f2468b = f2458b;
        return this;
    }

    public final NotificationsQuery ofTypes(int... iArr) {
        this.f2459a.f2468b = (int[]) iArr.clone();
        return this;
    }

    public final NotificationsQuery withFilter(Filter filter, String str) {
        this.f2459a.f2470d = filter;
        this.f2459a.f2471e = str;
        return this;
    }

    public final NotificationsQuery withLimit(int i) {
        this.f2459a.f2467a = i;
        return this;
    }
}
